### PR TITLE
[RSDK-7629] Implement externally managed network

### PR DESCRIPTION
**Pull request by npmenard**

### CODE REVIEW

**Review by gvaradarajan — CHANGES_REQUESTED** [review]
The methodology looks good to me. Depending on the answers to my questions, it's possible no actual changes will be required.

File: micro-rdk/src/esp32/conn/network.rs [hunk 1]
```diff
@@ -156,3 +165,246 @@ impl Network for Box<BlockingEth<EspEth<'static, OpenEth>>> {
...
+        ev_hnd_arg: *mut std::ffi::c_void,
+        ev_base: esp_idf_svc::sys::esp_event_base_t,
+        ev_id: i32,
+        ev_data: *mut std::ffi::c_void,
+    ) {
+        let data: &mut Esp32ExternallyManagerNetworkInner = &mut *(ev_hnd_arg as *mut _);
+        let ev_id = ev_id as u32;
+        if ev_base == esp_idf_svc::sys::IP_EVENT {
+            // receiving an IP_EVENT is the only event that can transition us to connected state
+            if ev_id == esp_idf_svc::sys::ip_event_t_IP_EVENT_STA_GOT_IP
```
[the remark this answers is not in the document]
Comment (npmenard) [reply micro-rdk/src/esp32/conn/network.rs:383]: we would need to wrap the event in an `EspEvent` which is another level of indirection and not suitable for this purpose. I would rather use the rust wrapper as is but that won't be possible with the current lib.

File: micro-rdk/src/esp32/conn/network.rs [hunk 2]
```diff
@@ -156,3 +165,246 @@ impl Network for Box<BlockingEth<EspEth<'static, OpenEth>>> {
...
+        // 2) confirm the loop is already instantiated (ESP_ERR_INVALID_STATE)
+        // Any other errors are fatal
+        let hnd = unsafe { esp_idf_svc::sys::esp_event_loop_create_default() };
+        if hnd != 0 && hnd != esp_idf_svc::sys::ESP_ERR_INVALID_STATE {
+            panic!("esp default event loop cannot be instantiated")
+        }
+
+        let mut this = Self { inner: data };
+
+        if let Err(err) = unsafe {
```
[the remark this answers is not in the document]
Comment (gvaradarajan) [reply micro-rdk/src/esp32/conn/network.rs:338]: Ok, that makes sense

File: micro-rdk/src/esp32/conn/network.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (gvaradarajan) [reply micro-rdk/src/esp32/conn/network.rs:383]: Also makes sense

**Review by gvaradarajan — APPROVED** [review]
LGTM

**Pull request merged**